types/mongodb/index.d.ts

```diff
@@ -463,26 +461,25 @@ export interface Collection {
     dropIndexes(): Promise<any>;
     dropIndexes(callback?: MongoCallback<any>): void;
     //http://mongodb.github.io/node-mongodb-native/2.1/api/Collection.html#find
-    find(query?: Object): Cursor<any>;
-    find<T>(query?: Object): Cursor<T>;
+    find(query?: Object): Cursor<TSchema>;
     /** @deprecated */
-    find(query: Object, fields?: Object, skip?: number, limit?: number, timeout?: number): Cursor<any>;
+    find<T = TSchema>(query: Object, fields?: Object, skip?: number, limit?: number, timeout?: number): Cursor<T>;
```

Comment:
Yes, When `fields` parameter was passed in, the results would not have some properties which `TSchema` has. This is an old api [http://mongodb.github.io/node-mongodb-native/1.4/api-generated/collection.html#find](http://mongodb.github.io/node-mongodb-native/1.4/api-generated/collection.html#find )